clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -747,11 +747,14 @@ public boolean rejoinNeededOrPending() {
 
         // we need to rejoin if we performed the assignment and metadata has changed;
         // also for those owned-but-no-longer-existed partitions we should drop them as lost
-        if (assignmentSnapshot != null && !assignmentSnapshot.matches(metadataSnapshot))
+        if (assignmentSnapshot != null && !assignmentSnapshot.matches(metadataSnapshot)) {
+            requestRejoin();
```

Comment:
Pardon me, I failed to catch the fixing/improvement about setting  flag ```rejoinNeeded``` to true.
The flag ```rejoinNeeded``` is read by ```AbstractCoordinator#rejoinNeededOrPending``` only and the return value is set to ```true``` in this overrided method (```ConsumerCoordinator#rejoinNeededOrPending```). Hence, setting flag ```rejoinNeeded``` to true seems to an NOP?

Reply:
It doesn't affect the first time this path is taken. We set the flag to ensure that if join fails,  we retry.

Reply:
Got it. thanks for explanation!